### ISSUE TITLE
Header title overlap back arrow

### ISSUE THREAD

**Issue by rutvikwebmobtech** [react-navigation/react-navigation#4754]
### My code 
     const AppRoute = createStackNavigator({

            Login: {
                screen: Login,
                navigationOptions: {
                    title: 'Login'
                },
            },
            Home: {
                screen: Home,
                navigationOptions: {
                    title: 'Home',
                    headerStyle:{
                        backgroundColor:'yellow'
                    }
               },
            }
        },
        {
            initialRouteName: 'Login',
        }
    );

### Output

![screenshot_1532411065](https://user-images.githubusercontent.com/31410282/43119314-901c3af8-8f33-11e8-8ed4-007843376bad.png)

### My Environment

| software         | version
| ---------------- | -------
| react-navigation | 2.9.0
| react-native     | 0.56.0
| node             | 8.10.0
| npm or yarn      | 5.10.0

**Comment by simpleblack** [comment]
The same problem

software | version
-- | --
react-navigation | 2.9.0
react-native | 0.55.4
node | 8.10.0
npm or yarn | 5.10.0

**Comment by Dimous** [comment]
It is the opposite in my case. A big left margin appeared.
It seems like devs mixed up styles in 12b21f052e73a5698138c3f37d775e87098eca00

**Comment by Dimous** [comment]
```typescript
public static navigationOptions: NavigationStackScreenOptions = {
        title: "Бюджет",
        headerRight: <View
            style={
                     {
                        flex: 1,
                        flexDirection: "row",
                        alignItems: "center",
                        justifyContent: "space-between",
                        width: 28+28+10,
                        marginRight: 10,
                    }
            }
        >
            <TouchableOpacity
                onPress={
                    () => {}
                }
            >        
                <Feather
                    name="filter"
                    size={28}
                    color="#000000"
                />
            </TouchableOpacity>
            <TouchableOpacity
                onPress={
                    () => {}
                }
            >            
                <Feather
                    name="plus"
                    size={28}   
                    color="#000000"                
                />
            </TouchableOpacity>
        </View>,
        headerStyle: {
            elevation: 1,
            marginBottom: 1,
            backgroundColor: "#fafafa",            
        },
        headerTitleStyle: {
            fontWeight: "200",
            fontFamily: "Roboto",
        },        
        headerTransparent: true,
        headerTitleAllowFontScaling: true,
    };
```
![screenshot_1532432181](https://user-images.githubusercontent.com/283939/43135724-b736e998-8f89-11e8-9ff0-fb8db3a658dc.png)

**Comment by Ariel-Rodriguez** [comment]
I am facing same problem

![image](https://user-images.githubusercontent.com/464285/43148242-493ba256-8f3b-11e8-89e0-7edb44d88c9c.png)

![image](https://user-images.githubusercontent.com/464285/43148260-5231f040-8f3b-11e8-878c-1efb41164122.png)

Sharing my work around to avoid this, until this issue gets solved. (However, margins are still wrong)

```javascript
  static navigationOptions = {
    title: 'Seguidores',
    // Empty view to avoid small bug https://github.com/react-navigation/react-navigation/issues/4754
    headerRight: <View />,
  }
```

**Comment by brentvatne** [comment]
thanks @Ariel-Rodriguez for the info, I'll fix this right away

**Comment by rationalmachines** [comment]
Having the same issue as @Dimous & @Ariel-Rodriguez, happened when I upgraded from 2.3 release.

**Comment by brentvatne** [comment]
fixed in 2.9.1, thank you everybody for the info to help track it down!
